File: data/gpt-4o/readable_format_HPEC/cpp/file075.cpp
#include <stdio.h>
#include <omp.h>
int main()
{
 int var = 0;
#pragma omp parallel shared(var)
 {
#pragma omp single
 {
 var+=1;
 }
#pragma omp barrier
#pragma omp single
 {
 var+=1;
 }
 }
 printf ("var = %d\n", var);
 return 0;
}

File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file075.f90
program DRB120_barrier_orig_no
    use omp_lib
    implicit none

    integer :: var

    !$omp parallel shared(var)
    !$omp single
    var = var + 1;
    !$omp end single
    !$omp barrier

    !$omp single
    var = var + 1;
    !$omp end single
    !$omp end parallel

    print 100, var
    100 format ('var =',i3)
end program
